jsonnet/kube-prometheus/components/prometheus.libsonnet

```diff
@@ -326,6 +326,35 @@ function(params) {
       }],
     },
   },
+  
+  podMonitorKubeProxy: {
```

Comment:
Let's move it into `k8s-control-plane.libsonnet` as it doesn't belong to prometheus, but to kubernetes.

Reply:
Ok but the name `k8s-**control-plane**.libsonnet` might be a little missleading ;)

Reply:
Agreed the name could be better. Right now it is the component handling kubernetes components (kubelet, apiserver, dns, etc...), so we might want to swap it to `kubernetes.libsonnet` in the future. However, this would need an assessment of how much breaking change this would be and if it is worth it (so definitely not as part of this PR :slightly_smiling_face:  ).